src/common/turbo.ts

```diff
@@ -90,28 +138,72 @@ export interface ArNSPaymentProvider {
     winc: string;
     mARIO: mARIOToken;
   }>;
+}
+
+export interface ArNSAuthenticatedPaymentProvider extends ArNSPaymentProvider {
   initiateArNSPurchase(
     params: AoTokenCostParams & { processId?: TransactionId },
-    options: WriteOptions,
+    options: WriteOptions, // Note: WriteOptions might not be necessary here if unused
   ): Promise<AoMessageResult<ArNSPurchaseReceipt>>;
 }
 
-export class TurboArNSPaymentProvider implements ArNSPaymentProvider {
-  private readonly paymentUrl: string;
-  private readonly axios: AxiosInstance;
-  private readonly logger: ILogger;
-  private readonly signer?: TurboArNSSigner;
+export class TurboArNSPaymentFactory {
```

Comment:
TurboPaymentProvider
```suggestion
export class TurboPaymentProvider {
```